docs/release.md

```diff
@@ -7,27 +7,6 @@ This document describes how to release a new version.
 
 Follow [semantic versioning 2.0.0][semver] to choose the new version number.
 
```

Comment:
I thought it would be helpful to have an explanation that release notes are made based on the title of the PR and the label attached to it.